openmdao/core/group.py

```diff
@@ -1661,21 +1662,22 @@ def _setup_connections(self):
             in_units = allprocs_abs2meta_in[abs_in]['units']
 
             if out_units:
-                if not in_units:
+                if not in_units and _is_unitless(out_units):
                     msg = f"{self.msginfo}: Output '{abs_out}' with units of '{out_units}' " + \
                           f"is connected to input '{abs_in}' which has no units."
                     simple_warning(msg)
-                elif not is_compatible(in_units, out_units):
+                elif _is_unitless(out_units) and not is_compatible(in_units, out_units):
```

Comment:
Don't we still need a check for the case where out_units is NOT unitless but is not compatible with in_units?  I'm not sure why the _is_unitless check is even necessary here.  Shouldn't is_compatible handle this?